### ISSUE TITLE
nvim not found when using "neovide --wsl" in powershell (windows 10)

### ISSUE THREAD

**Issue by tianhao21** [neovide/neovide#2778]
I was trying to use neovide with wsl in win10, but it doesn't work
i used `neovide --wsl` and it said nvim not found
![image](https://github.com/user-attachments/assets/a006ec27-354e-44c4-9cf7-199e0c548b70)

I have set my wsl as default
![image](https://github.com/user-attachments/assets/f4ee07a7-a030-4930-843c-45b3704e1812)

I have installed neovim following [this](https://github.com/neovim/neovim/blob/master/INSTALL.md#linux)
![image](https://github.com/user-attachments/assets/f2ca299f-b216-4c1e-b6f2-b725814bb61e)
![image](https://github.com/user-attachments/assets/5f2123db-f530-488a-8154-d3090217116c)

but when i tried `neovide --wsl --neovim-bin /opt/nvim-linux64/bin/nvim`, it works

what's the reason and how can i fix it?

**Comment by Konstantin-Roumas** [comment]
I have the same issue on Mac M1

**Comment by Konstantin-Roumas** [comment]
<img width="512" alt="image" src="https://github.com/user-attachments/assets/b9248e11-c957-4025-a514-b8cd665d0263">
When i want to launch from Finder of Desktop

**Comment by fredizzimo** [comment]
This might be fixed by this https://github.com/neovide/neovide/pull/2950, are you able to test that? You can download the binaries from the CI https://github.com/neovide/neovide/actions/runs/12703393292

**Comment by GreenHandHand** [comment]
I ran into the same problem while using Neovide in WSL2. Checking the logs, I saw that Neovide was trying to find nvim with this command:
```shell
> wsl zsh -c 'which nvim'
nvim not found
```
But it couldn't locate nvim. However, when I run which nvim directly inside WSL, it works just fine and returns the path:
```shell
> which nvim
/opt/nvim-linux64/bin/nvim
```
I realized the issue might be because I add the nvim path to my environment variables using export in my .zshrc file like this:
```shell
export PATH="$PATH:/opt/nvim-linux64/bin"
```
So, when Neovide runs its command from Windows, it doesn't load my `.zshrc`, which means it doesn't see the updated `PATH` where nvim is located. To fix this, the command should first load my `.zshrc` settings before looking for nvim:
```shell
wsl zsh -c 'source ~/.zshrc && which nvim'
```
This explains why which nvim works inside WSL but not when running the command from Windows.

To get everything working, I added these lines to my .zshrc:
```shell
// In your .bashrc or .zshrc
export PATH="$PATH:/opt/nvim-linux64/bin"
alias neovide="/mnt/c/Program\ Files/Neovide/neovide.exe --wsl --neovim-bin $(which nvim)"
```
Now Neovide starts up without any issues. Hope this helps someone else facing the same problem!

**Comment by fredizzimo** [comment]
@GreeenHandHand, we still needmto improve our FAQ here
https://neovide.dev/faq.html#neovide-is-not-picking-up-some-shell-configured-information
And here https://neovide.dev/faq.html#macos-login-shells (This section mostly applies to WSL too)

But, in short as you noticed, `.zshrc` is not read because Neovide does not launch an interactive shell, just a non-interactive login shell. `.zprofile` and `.zshenv` are read, so if you move your path exports there it should be fine. The cannonica location for defining paths are in the `.zshenv` file (on macOS the `.zprofile` file has to be used due to apples strange implementation)

**Comment by GreenHandHand** [comment]
@fredizzimo Yes, exactly as you said. After moving the `export PATH="$PATH:/opt/nvim-linux64/bin"` command from `.zshrc` to `.zshenv`, Neovide was able to find nvim properly within WSL. The FAQ explains this issue in great detail.

---

As a side note, the installation instructions for nvim recommend adding the export command to `.bashrc` or `.zshrc`, which works perfectly fine in Linux. However, when launching Neovide via `neovide.exe --wsl` from Windows CMD, it leads to the problem mentioned above.

![Image](https://github.com/user-attachments/assets/6ea1f257-da87-43f1-b974-018c0ee0af81)

It's definitely a tricky issue if you're not aware that it's related to how `.zshrc` and `.zshenv` are loaded. When I first looked through the FAQ, I didn't realize the problem was tied to the `.zshrc`. Thank you very much!

**Comment by fredizzimo** [comment]
> As a side note, the installation instructions for nvim recommend adding the export command to `.bashrc` or `.zshrc`, which works perfectly fine in Linux. However, when launching Neovide via `neovide.exe --wsl` from Windows CMD, it leads to the problem mentioned above.

Yes, that's unfortunate, most similar instructions out there almost always recommend `.bashrc` or `.zshrc`. Probably because of simlicity, and because everyone else does it. It also has the advantage of taking effect immediately, you just need to restart the terminal emulator or open a new shell, while updating `.zshrc` requires a complete logout, or other trickery.

Zsh itself recommends `.zshenv` https://zsh.sourceforge.io/Guide/zshguide02.html#l23.
